lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -84,6 +84,36 @@ ::mlir::LogicalResult mlir::tt::ttnn::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttnn::ReshapeOp::verify() {
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  ArrayAttr shape = getShape();
+  int64_t shape_size = static_cast<int64_t>(shape.size());
+
+  if (shape_size == 0) {
+    return emitOpError("Shape attribute must be non-empty");
+  }
+
+  if (shape_size > 5) {
+    return emitOpError("Shape attribute must have at most 5 elements");
+  }
+
+  if (shape_size != outputType.getRank()) {
```

Comment:
Should we also check if the reshaped volume is the same as original tensor's volume?

Reply:
Yea, probably. I already pinged metal forks to send me a list of all constraints. Once I get them all I will add them.

Reply:
Added, tnx!